### ISSUE TITLE
[BG-Alterac Valley] Mine Supplies never despawn

### ISSUE THREAD

**Issue by Nyeriah** [azerothcore/azerothcore-wotlk#8548]
### Current Behaviour

Original CC Report: https://github.com/chromiecraft/chromiecraft/issues/2096

Irondeep supplies (https://wowgaming.altervista.org/aowow/?object=178785) in Alterac Valley can be repeatedly looted.
They are for quest Irondeep Supplies (https://wowgaming.altervista.org/aowow/?object=178785) which awards 3g 10 silver and can be done repeatedly.

https://user-images.githubusercontent.com/80769357/137381185-eccde730-cf94-4362-a309-5dca8f4fa2a9.mp4

### Expected Blizzlike Behaviour

After looting Irondeep Supplies they should despawn, and go on respawn timer.

### Source

https://youtu.be/nwhM9Ari434?t=8

### Steps to reproduce the problem

1. Go into Alterac Valley
2. take quest Irondeep Supplies (https://wowgaming.altervista.org/aowow/?object=178785)
3. Go to Irondeep Mine and loot the supplies repeatedly

### Extra Notes

**Nyeriah:**
Has to do with how BG gameobjects are handled, they're presumed to never despawn (fine for most battlegrounds, but AV is an exception as it has PvE elements like this)

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/b2f08934bcc60491f8fa8971f98837b2d48a489e

### Operating system

Win11 x64

**Comment by Nyeriah** [comment]
Some important research notes that I will leave here so they don't get lost:

GameObjects spawned are created without a respawn timer (in general) https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/server/game/Entities/GameObject/GameObject.cpp#L250

The respawntimer tends to be filled after the object has been created, during the spawn process.

But this does not happen in battlegrounds: https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/server/game/Battlegrounds/Zones/BattlegroundAV.cpp#L1272

Notice that there's a RESPAWN_ONE_DAY param being passed into that function call. Turns out that param has been silenced: https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/server/game/Battlegrounds/Battleground.cpp#L1511 

In other words, despite being passed, it isn't used for anything.

The default respawn timer for GameObjects in battlegrounds seems to be set to one day. I added logs everywhere, wasn't able to figure out where that is set.

However, the respawnDelayTimer seems to always be 0, which causes them to instantly respawn right after being used. Setting the respawnDelayTimer causes them to behave properly.
